Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabInstanceUtils.cpp

```diff
@@ -14,46 +16,94 @@ namespace AzToolsFramework
-                InstanceOptionalConstReference instance = *startInstance;
-                AZStd::vector<InstanceOptionalConstReference> instancesInHierarchy;
+                result.reachedInstance = *instance;
+                return result;
+            }
+
+            AZStd::string GetRelativePathBetweenInstances(const Instance& parentInstance, const Instance& childInstance)
+            {
+                const Instance* instance = &childInstance;
+                AZStd::vector<InstanceOptionalConstReference> climbedInstances;
+
+                // Climbs up the instance hierarchy from the child instance until it hits the parent instance.
+                while (!CompareInstances(*instance, parentInstance) && instance->HasParentInstance())
+                {
+                    climbedInstances.emplace_back(*instance);
+                    instance = &(instance->GetParentInstance()->get());
+                }
 
-                while (&instance->get() != targetInstance && instance->get().GetParentInstance() != AZStd::nullopt)
+                if (!CompareInstances(*instance, parentInstance))
                 {
-                    instancesInHierarchy.emplace_back(instance);
-                    instance = instance->get().GetParentInstance();
+                    AZ_Warning("Prefab", false, "PrefabInstanceUtils::GetRelativePathBetweenInstances() - "
+                               "Tried to get relative path but the parent instance is not a valid parent. Returns empty string instead.");
+                    return "";
                 }
 
-                return AZStd::make_pair(&instance->get(), AZStd::move(instancesInHierarchy));
+                // Generates relative path.
+                return GetRelativePath(climbedInstances);
             }
 
-            AZStd::pair<const Instance*, AZStd::string> GetRelativePathBetweenInstances(
-                const Instance* startInstance, const Instance* targetInstance)
+            AZStd::string GetRelativePath(const AZStd::vector<InstanceOptionalConstReference>& climbedInstances)
             {
-                if (!startInstance)
+                AZStd::string relativePath = "";
+
+                for (auto instanceIter = climbedInstances.rbegin(); instanceIter != climbedInstances.rend(); ++instanceIter)
                 {
-                    return AZStd::make_pair(nullptr, "");
+                    relativePath.append(PrefabDomUtils::PathStartingWithInstances);
+                    const Instance& instance = (*instanceIter)->get();
+                    relativePath.append(instance.GetInstanceAlias());
                 }
 
-                auto climbUpToTargetOrRootInstanceResult = ClimbUpToTargetOrRootInstance(startInstance, targetInstance);
+                return relativePath;
+            }
+
+            bool IsHierarchical(const Instance& childInstance, const Instance& parentInstance)
```

Comment:
Based on the header comments, it said that it checks for both ancestor and descendant. But this is only checking it one way. Should both 'IsHierarchical' and 'IsProperlyHierarchical' be merged into one?

Reply:
The intent is check if child instance is descendent of a parent instance. If it is true, the parent is also the ancestor. I think they are the same thing. But I would update the comment and the name to indicate it checks either ancestor or descendant instead of mentioning both.

Reply:
> Should both 'IsHierarchical' and 'IsProperlyHierarchical' be merged into one?

```
Car
|-- Wheel

// IsHierarchical(Wheel, Wheel) -> True
// IsHierarchical(Wheel, Car) -> True

// IsProperlyHierarchical(Wheel, Wheel) -> False
// IsProperlyHierarchical(Wheel, Car) -> True
```
We can remove `IsProperlyHierarchical` and explicitly specify a "properly" check on input in `IsHierarchical`.

Reply:
Okay. It can just be one function then and comments need to be updated. Because based on the comments I read it as
```
IsHierarchical(Wheel, Wheel) -> True
IsHierarchical(Car, Wheel) -> True
```